pdns/speedtest.cc

```diff
@@ -1181,6 +1182,47 @@ struct SipHashTest
 };
 #endif
 
+struct DedupRecordsTest
+{
+  explicit DedupRecordsTest(size_t howmany, bool dedup, bool withdup = false) : d_howmany(howmany), d_dedup(dedup), d_withdup(withdup)
+  {
+  }
+
+  [[nodiscard]] string getName() const
+  {
+    return std::to_string(d_howmany) + " DedupRecords" + std::string(d_dedup ? "" : " (generate only)") +
+      std::string(d_withdup ? " (with dup)" : "");
+  }
+
+  void operator()() const
+  {
+    std::vector<DNSRecord> vec;
+    vec.reserve(d_howmany);
+    std::string name("some.name.in.some.domain");
+    auto count = d_howmany;
+    if (d_withdup) {
+      count--;
+    }
+    for (size_t i = 0; i < count; i++) {
+      auto content = DNSRecordContent::make(QType::TXT, QClass::IN, "\"a text " + std::to_string(i) + "\"");
+      DNSRecord rec(name, content, QType::TXT);
+      if (i == 0 && d_withdup) {
+        vec.emplace_back(rec);
+      }
+      vec.emplace_back(std::move(rec));
+    }
```

Comment:
Would it make sense to do this in the constructor instead, to get the initial steps out of the timed computation?